### ISSUE TITLE
[Bug] No menu item "insert items using templates"  (ReferenceError: topItem is not defined) 

### ISSUE THREAD

**Issue by andersbc** [windingwind/zotero-better-notes#553]
**Describe the bug**
I am trying to use a template that summarises meta data information about a zotero item (e.g. a reference), but I do not have the menu item edit -> "insert items using templates" which is shown in the README.md 

**README.md gif: (shows edit -> "insert items using templates")**
<div><img src="https://user-images.githubusercontent.com/33902321/236623159-8f67064b-1fab-4cf1-abf4-0c8243370a14.gif" width="600px"></img></div>

**My menu: (no "insert items using templates" option)**
<div><img src="https://github.com/windingwind/zotero-better-notes/assets/1403483/d6c15b81-5f17-4b6c-b31e-c29b2ef8d730" width="200px"></img></div>

 - OS: WIN11
 - Zotero: 6.0.26
 - Better Notes: 1.0.4

**Additionally - the documentation on templates are not very helpful on how to generate metadata**

This is my simple template

```html
<h2>Item Meta Data:</h2>
<p>Title: ${topItem.getField("title")}</p>
```

When I tried to use the template inside a existing note ('insert template to Cursor line'), I get `Template Error: ReferenceError: topItem is not defined`

I know javascript/html, etc. 

..but the template doc:  
https://github.com/windingwind/zotero-better-notes/blob/master/docs/about-note-template.md#write-note-template

.. does not give me any helpful  on what the `topItem` IS? (it is used in many templates) 
for instance why don't I have access to it inside a note as my error message implies? I would expect it to reference the "parent" of the note, e.g. the zotero reference that the note is nested under..  

maybe the docs could could be a little more clear in what it is.. is it a global variable always accessible to the template, or? can it be used inside an existing nested note? etc.

Thank you

**Comment by windingwind** [comment]
Item is the paper, attachment, etc. in your Zotero library. `topItem` is the item you select in the item picker window when calling a `[Item]` template. The template name must starts with `[Item]`.

**Comment by windingwind** [comment]
The documentation is outdated, as pointed out in the readme. For the 1.0.0+ version, reading the readme would be enough. If there is something that is not clearly expressed in the readme, please let me know.

**Comment by andersbc** [comment]
Thanks for the reply.. 

OK - I see the mention of Item in https://github.com/windingwind/zotero-better-notes/blob/master/docs/about-note-template.md#note-template-structure, thnaks for pointing that out

What is unclear to me, is how to apply an "item template". 

Unlike the gif in the README.md I don't have the "insert items using templates" menu item in my "edit" menu. 
I don't know how else to get to the "item picker window"? .

..is this window built-in to Zotero or something provided by the plugin?

Thanks

**Comment by windingwind** [comment]
It's automatically detected what kind of a template you are using when selecting the template. If it is an item template, the item picker shows.

**Comment by andersbc** [comment]
Thanks for the reply
 
I guess I don't know how to select the item template then: 

**a) from main window**
...see my screenshot in the first post, there is no "insert items using template" option in the edit menu.

**b) inside an existing note:**
If I try to insert the item template in an existing note (nested under a reference), by clicking gear -> "insert template to cursor line" and select the item template, it just comes up with `Template Error: ReferenceError: topItem is not defined`

![zotero_a5Np0iEKH3](https://github.com/windingwind/zotero-better-notes/assets/1403483/dee36589-4637-40ed-840b-b21d51454d0d)

This is the item template I select in the gif above

![Screenshot 2023-07-05 175404](https://github.com/windingwind/zotero-better-notes/assets/1403483/1279b5fa-5401-489f-86b9-8e9596e8774b)

Any ideas on what is wrong?
Thanks

**Comment by andersbc** [comment]
OK, I figured it out - the template should actually be named "`[Item] ... `"  .. then it works:

![Screenshot 2023-07-05 175404](https://github.com/windingwind/zotero-better-notes/assets/1403483/85a33269-e400-49c4-9214-f916ab42261f)


.. apparently It is not sufficient that the actual template code starts with  `name: "[Item] Example Item Template" ` 

This is a little confusing -  maybe I didn't read the docs well enough -  but I would think that the code in the template is what matters ...it seems counter intuitive / redundant that you must specify the name two places, but if this is the case, maybe it could be clarified or extra highlighted in the docs?

Thanks for the plugin :-)

**Comment by windingwind** [comment]
Well, there seems to be some misunderstanding.

1. The template code you showed in the screenshot is for `Import from clipboard`. Directly copy-paste it into the editor does not create what the template was designed to do
2. If you want to manually paste it, please copy the string after `name: ` as the template name and copy the lines after `content: ` as the real template code.
3. 
> insert items using template is missing


As I mentioned above:

> It's automatically detected what kind of a template you are using when selecting the template. If it is an item template, the item picker shows. 

Now you don't need to manually select what type of template (item, or something else) is. Use insert template directly, select the template you want, that's all.

**Comment by windingwind** [comment]
> .. apparently It is not sufficient that the actual template code starts with name: "[Item] Example Item Template" 

That's not how it works. Please read the steps to use the template, which is supposed to be placed near the place you copied that `template for importing` code.

> This is a little confusing - maybe I didn't read the docs well enough - I would think that the code in the template is what matters ...it seems counter-intuitive / redundant that you must specify the name two separate places, but if this is the case, maybe it could be clarified or extra highlighted in the docs?

No, template name starting with `[item]` is enough.

**Comment by windingwind** [comment]
Well, I go back to the doc and found that it was not explicitly marked as `for importing`. My bad. Thanks for pointing this out.

I have updated the doc and now it(and all other for-importing template code should be like this:

```yaml
# This template is for importing.
# Do not copy-paste this to the editor directly.
# To import the template, copy the content and
# goto Zotero menubar, click Edit->New Template from Clipboard.
name: "[Text] Current Time"
content: |-
  // @use-markdown
  // @author windingwind
  // @link https://github.com/windingwind/zotero-better-notes/blob/master/docs/about-note-template.md

  # Template Example: Current Time

  > Author: windingwind
  >
  > from: [GitHub-Zotero Better Notes: Write Note Template](https://github.com/windingwind/zotero-better-notes/blob/master/docs/about-note-template.md)

  **Current Time**: ${new Date().toLocaleString()}
```

Is it better now?

**Comment by andersbc** [comment]
Thank you for your comments and update to the docs.

Yes, much better!  

However, maybe even more clarification? 

``` 
# This template is specifically for importing/sharing, using better 
# notes 'import from clipboard': copy the content and
# goto Zotero menu bar, click Edit->New Template from Clipboard.  
# Do not copy-paste this to better notes template editor directly.
```

- editor -> "better notes template editor" (to not confuse with the note editor or other places you can edit things)

... and maybe, when you have the time, add a section to the docs on the "template editor", maybe with a link just below the snippet above: "See here, if you want to edit /write templates directly in better notes template editor" 

Essentially, the template in the editor is just an 'unwrapped' version of template.  

I think it would be great to highlight that more because the intuition is that "a template is just a template", users do not suspect that there are two kinds of template formats - one for the editor and one for sharing. My guess is that the template editor maybe is a new feature, so it has not been documented yet.

Thanks!

**Comment by windingwind** [comment]
Thanks, I've updated the comment. The template editor is not new, but the initial place for adding new templates. However, many users don't know how to use that and I introduced the `import from clipboard` in the 1.0.0+.

It is confusing there are two types of templates, but actually, there aren't. The old templates users share in the discussions are the template's content, not the template itself. A template consists of a name and content, but they usually ignore the name and recognize the content as a template. Since the introduction of `import`, I encourage new contributors to provide the entire template that is for share and import.
